lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -20,13 +20,17 @@ function wp_theme_has_theme_json() {
 		$theme_has_support = wp_cache_get( $cache_key, $cache_group );
 
 		/**
-		 * $theme_has_support is stored as a int in the cache.
+		 * $theme_has_support is stored as an int in the cache.
 		 *
 		 * The reason not to store it as a boolean is to avoid working
 		 * with the $found parameter which apparently had some issues in some implementations
 		 * https://developer.wordpress.org/reference/functions/wp_cache_get/
 		 */
-		if ( 0 === $theme_has_support || 1 === $theme_has_support ) {
+		if (
+			// Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
+			( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
```

Comment:
How about using `wp_get_environment_type()` function here?

Reply:
@spacedmonkey That's another great alternative idea we should consider, cc @mmtr @oandregal.

Can we open an issue to introduce a dedicated function for this kind of check? In such an issue we can discuss which approach we want to take. I hadn't thought about `wp_get_environment_type()`, but I like that a lot as a baseline for this, e.g. we could bypass cache if the environment is `development` or `local`.

Reply:
Opened #45912

Reply:
Good suggestion indeed! Happy to explore it separately in https://github.com/WordPress/gutenberg/issues/45912

Reply:
Just want to note that it's not possible that `WP_DEBUG` is not defined so there's no need to always check for `defined( 'WP_DEBUG' )`. `wp_initial_constants()` is responsible for that.